src/parallel/pipeline.cpp

```diff
@@ -96,6 +96,8 @@ void Pipeline::ScheduleSequentialTask() {
 
 bool Pipeline::ScheduleOperator(PhysicalOperator *op) {
 	switch (op->type) {
+	case PhysicalOperatorType::CREATE_TABLE_AS:
```

Comment:
`CREATE_TABLE_AS` should not be here, it should only be parallelized as a sink. To maybe clarify a bit better what this is doing here:

Pipelines have two schedule functions, `Schedule` and `ScheduleOperator`. `Schedule` looks at the first node (the sink) and, if that sink can be parallelized, moves up the pipeline by calling `ScheduleOperator`. In `ScheduleOperator`, the nodes are **not** sinks but either (1) a source node, or (2) an intermediate operator. For example, the hash join here is the **hash probe**, not the hash build (the hash build is when the hash join is a sink). 

When this is called with a `CREATE_TABLE_AS` node as input, the `CREATE_TABLE_AS` is a **source**, i.e. this is the phase when `GetChunk` is called on the node. This should never be parallelized, as `CREATE_TABLE_AS` only returns a single value on `GetChunk` (the count). The parallelism factor here also does not depend on the children of the `CREATE_TABLE_AS` node, since they are not involved anymore (they are only involved when the node is a sink, which is handled in the `Schedule` function below).

In practice this doesn't make a difference since `CREATE_TABLE_AS` is generally the root node of the plan, and the root of the plan will never be parallelized, but the distinction is important and could potentially lead to bugs later on if this were to change (e.g. if we were to make plans that create multiple tables at once).

Reply:
Thanks for the clarification! The example with building/probing a hash join clears it up for me. I've removed it here.